torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1885,6 +1918,23 @@ def _simplify_floor_div(self, expr):
             self.evaluate_expr(eq_expr)
         return self.simplify(expr)
 
+    def _add_guard(self, expr: "sympy.Expr") -> None:
+        stack = get_debugging_stack()
+        guard = ShapeGuard(expr, stack)
+        if torch._dynamo.config.print_guards:
+            # reusing flag that prints guards
+            frame_summaries = TracingContext.get().frame_summary_stack
+            # frame_summaries describes a stack of functions
+            # TODO(avik): It would be better to describe a stack of function calls instead
+            current_loc = TracingContext.get().loc_in_frame
+            # current_loc describes a line in the current frame
+            # TODO: optimize this; avoid formatting traces until we need them
+            user_stack = ''.join(traceback.format_list([*frame_summaries, current_loc]))
+            expr = LoggingShapeGuardPrinter(self.var_to_sources).doprint(expr)
+            torch._dynamo.guards.log.warning(f"Adding shape guard {expr} at \n{user_stack}")
```

Comment:
I think the easiest way to avoid useless string formatting when it is not necessary is to test the log level and only do the string formatting message when it is high enough.

Reply:
done